packages/ui/src/components/Menu/SSubmenu.vue

```diff
@@ -0,0 +1,129 @@
+<script lang="ts">
+export default { name: 'SSubmenu' }
+</script>
+
+<script setup lang="ts">
+import SMenuItemBody from '@/components/Menu/SMenuItemBody.vue'
+import { SCollapseTransition } from '@/components/Transitions'
+import { SUBMENU_API_KEY, useMenuApi } from '@/components/Menu/api'
+import { IconChevronBottom16 } from '@/components/icons'
+
+const includedItems = shallowReactive(new Set<string>([]))
+
+const api = {
+  register,
+  unregister,
+}
+provide(SUBMENU_API_KEY, api)
+
+function register(index: string) {
+  includedItems.add(index)
+}
+
+function unregister(index: string) {
+  includedItems.delete(index)
+}
+
+const [opened, toggle] = useToggle(false)
+
+function handleTriggerClick() {
+  if (!isMenuCollapsed.value) {
+    toggle()
+  }
+}
+
+let isMenuCollapsed = ref(false)
+let hasActiveItem = ref(false)
+const menuApi = useMenuApi()
+
+if (menuApi) {
+  isMenuCollapsed = computed(() => menuApi.collapsed.value)
+  hasActiveItem = computed(() => includedItems.has(menuApi.active.value))
+}
+
+watch(hasActiveItem, () => {
+  if (!isMenuCollapsed.value && hasActiveItem.value) {
+    toggle(true)
+  }
+})
+
+watch(isMenuCollapsed, () => {
+  if (isMenuCollapsed.value) {
+    toggle(false)
+  }
+})
```

Comment:
`hasActiveItem` and `isMenuCollapsed` could be initially `true` (if `menuApi` exists). Thus, these watchers should be `immediate: true`.

Reply:
Also, they could be rewritten with `whenever`:

```ts
import { and, not } from '@vueuse/shared'

whenever(and(not(isMenuCollapsed), hasActiveItem), () => toggle(true), { immediate: true })

whenever(isMenuCollapsed, () => toggle(false), { immediate: true })
```